apps/server/src/modules/group/repo/group.repo.ts

```diff
@@ -98,4 +157,67 @@ export class GroupRepo extends BaseDomainObjectRepo<Group, GroupEntity> {
 
 		return domainObjects;
 	}
+
+	private async findAvailableGroup(
+		pipelineStage: unknown[],
+		skip = 0,
+		limit?: number,
+		nameQuery?: string
+	): Promise<Page<Group>> {
+		let nameRegexFilter = {};
+		const pipeline: unknown[] = pipelineStage;
+
+		if (nameQuery && StringValidator.isNotEmptyString(nameQuery, true)) {
+			const escapedName = nameQuery.replace(MongoPatterns.REGEX_MONGO_LANGUAGE_PATTERN_WHITELIST, '').trim();
+			if (StringValidator.isNotEmptyString(escapedName, true)) {
+				nameRegexFilter = { name: { $regex: nameQuery, $options: 'i' } };
+			}
+		}
+
+		pipeline.push(
+			{ $match: nameRegexFilter },
+			{
+				$lookup: {
+					from: 'courses',
+					localField: '_id',
+					foreignField: 'syncedWithGroup',
+					as: 'syncedCourses',
+				},
+			},
+			{ $match: { syncedCourses: { $size: 0 } } }
+		);
+
+		if (limit) {
+			pipeline.push({
+				$facet: {
+					total: [{ $count: 'count' }],
+					data: [{ $skip: skip }, { $limit: limit }],
+				},
+			});
+		} else {
+			pipeline.push({
+				$facet: {
+					total: [{ $count: 'count' }],
+					data: [{ $skip: skip }],
+				},
+			});
+		}
+
+		const mongoEntities = await this.em.aggregate(GroupEntity, pipeline);
```

Comment:
mongoEntities is not a good name for the object that you get from the facet

Reply:
this might also be a rare case where an `as` cast makes more sense than a linter suppression.

Reply:
`as [{ total: [{ count: number }], data: GroupEntity[] }]`